src/App.tsx

```diff
@@ -15,36 +16,101 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortBy {
+  None,
+  Abc,
+  Length,
+}
+
+enum Reverse {
+  No,
+  Yes,
+}
+
+const sortGoods = (sortField: SortBy, reverseField: Reverse) => {
+  const sortedGoods = [...goodsFromServer];
+
+  switch (sortField) {
+    case SortBy.Abc:
+      sortedGoods.sort((a, b) => a.localeCompare(b));
+      break;
+
+    case SortBy.Length:
+      sortedGoods.sort((a, b) => a.length - b.length);
+      break;
+
+    default:
+  }
+
+  if (reverseField) {
```

Comment:
The condition `if (reverseField)` will always evaluate to `true` because `reverseField` is an enum and any non-zero enum value is truthy. You should explicitly check for `Reverse.Yes` to determine if the list should be reversed.